pkgs/development/compilers/elm/packages/node-packages.json

```diff
@@ -1,10 +1,10 @@
 [
-    "elm-test",
-    "elm-verify-examples",
-    "elm-doc-preview",
-    "elm-upgrade",
+    { "elm-test": "0.19.0-rev6" },
+    { "elm-verify-examples": "4.0.2" },
+    { "elm-doc-preview": "3.0.1" },
+    { "elm-upgrade": "0.19.6" },
     { "elm-analyse": "0.16.3" },
-    "elm-live",
-    "elm-xref",
-    "@elm-tooling/elm-language-server"
+    { "elm-live": "4.0.1" },
+    { "elm-xref": "4.0.0" },
+    { "@elm-tooling/elm-language-server": "1.4.1" }
```

Comment:
just a question: Any reason why we need to pin those dependecies to these specific versions?

Reply:
from the commit message:

> I'm trying to upgrade a single Elm package version and can't without
> doing all of them. This pins the sources in NPM. Not perfect, but a
> start.

Is there a better way? I couldn't figure out which upgrades to npm deps were necessary for which upgrade and this is the only way I could figure out to separate them.

Reply:
so far I was always upgrading all of the dependecies and pinning only those which were failing. it's definitely a trade-off and we should consider pinning everything. My point would be that it's hard to keep track of new versions of every individual dependecy and it would make sense to try to keep as much up to date as possible by simply forcing upgrades when new stuff is added. Obvious downside is that it means you always need to test that everything is building when you add anything new but I belive `@GrahamcOfBorg ` would fail in case something can't be build after upgrade.

Reply:
I'm concerned that I can't follow the commit message guidelines when upgrading everything at once. I think in an ideal world we'd package all of these with separate node2nix invocations—then it doesn't have to be pinned and downstream dependencies updates are isolated.

Reply:
Yes you're right I was dealing with exactly the same issue. The best thing I come up with was to name PR something like `elmPackages.*: update packages`. Anyway I feel we need an some input from one of the commiters on this.

@domenkozar Can you point us to the right direction with this by any chance?

Reply:
Using a wildcard seems fine to me.

Reply:
So I would remove those versions and do updates of everything as a part of this PR. We can rename this to `elmPAckages.*:.....`. I will also try to build all the things myself locally to help with testing. Does it sound good @BrianHicks?

Reply:
I've built them all locally too—I'll squish it all down again and wildcard 'em. 😄